src/js/api/ExercisesService.js

```diff
@@ -0,0 +1,193 @@
+/**
+ * Service class to interact with exercises API.
+ */
+export default class ExercisesService {
+  #basePath = '/exercises';
+  #axios;
+  #page = 1;
+  #filter = '';
+  #name = '';
+
+  /**
+   * Creates an instance of ExercisesService.
+   * @param {Object} axios - Axios instance for making HTTP requests.
+   */
+  constructor(axios) {
+    this.#axios = axios;
+  }
+
+  /**
+   * Gets the current page number.
+   * @returns {number} The current page number.
+   */
+  get page() {
+    return this.#page;
+  }
+
+  /**
+   * Sets the current page number.
+   * @param {number} value - The new page number (must be a positive number).
+   */
+  set page(value) {
+    if (typeof value === 'number' && value >= 1) {
+      this.#page = value;
+    } else {
+      console.warn('Page must be a positive number. Setting to 1.');
+      this.#page = 1;
+    }
+  }
+
+  /**
+   * Gets the current filter string.
+   * @returns {string} The current filter string.
+   */
+  get filter() {
+    return this.#filter;
+  }
+
+  /**
+   * Sets the filter string.
+   * @param {string} value - The new filter string.
+   */
+  set filter(value) {
+    if (typeof value === 'string') {
+      this.#filter = value;
+    } else {
+      console.warn('Filter must be a string');
+    }
+  }
+
+  /**
+   * Gets the current name string.
+   * @returns {string} The current name string.
+   */
+  get name() {
+    return this.#name;
+  }
+
+  /**
+   * Sets the name string.
+   * @param {string} value - The new name string.
+   */
+  set name(value) {
+    if (typeof value === 'string') {
+      this.#name = value;
+    } else {
+      console.warn('Name must be a string');
+    }
+  }
+
+  /**
+   * Fetches exercises based on the current filters, name, and pagination.
+   * @returns {Promise<Object>} The data of exercises fetched from the API.
+   * @throws {Error} Throws an error if the request fails.
+   */
+  async getExercises() {
+    try {
+      const { data } = await this.#axios.get(
+        `${this.#basePath}?${
+          this.#filter && this.#name ? `${this.#filter}=${this.#name}&` : ''
+        }&page=${this.#page}&limit=10`
```

Comment:
We are going to use params for request
`axios.get(url, {params}`

This is redundant to get separate params from url search params object and then put it back to query string